src/util/basicUtil.ts

```diff
@@ -3,64 +3,109 @@ import { WMAIN_ADDRESS } from '../constants/contracts.json';
 import data from './data';
 import { log } from './logger/logger';
 import { config } from '../app.config';
+import { createClient } from 'redis';
 const DEFAULT_DELAY = 20;
+let client;
 
-function getPath(chain, provider) {
-  return `${config.BASE_URL}${chain}/${provider}`;
-}
-
-function getWmainAddress(chain) {
-  return WMAIN_ADDRESS[chain];
+async function redisInit() {
+  if (!client) {
+    console.log('init redis');
+    client = createClient({
+      url: 'redis://defi-providers-redis-01.qa.aws.dappradar.ns:6379',
+    });
+    await client.connect();
+  }
 }
 
-function getDelay(chain) {
-  return data.CHAINS[chain].delay || DEFAULT_DELAY;
-}
+function basicUtil() {
+  function getPath(chain, provider) {
+    return `${config.BASE_URL}${chain}/${provider}`;
+  }
 
-async function writeDataToFile(data, fileName, chain, provider) {
-  await new Promise<void>(function (resolve) {
-    fse.outputFile(
-      `${getPath(chain, provider)}/${fileName}`,
-      JSON.stringify(data, null, 2),
-      'utf8',
-      function (err) {
-        if (err) {
-          log.error({
-            message: err?.message || '',
-            stack: err?.stack || '',
-            detail: `Error: writeDataToFile`,
-            endpoint: 'writeDataToFile',
-          });
-        }
-        resolve();
-      },
-    );
-  });
-}
+  function getWmainAddress(chain) {
+    return WMAIN_ADDRESS[chain];
+  }
 
-function readDataFromFile(fileName, chain, provider) {
-  console.log(`${getPath(chain, provider)}/${fileName}`);
-  return JSON.parse(
-    fse.readFileSync(`${getPath(chain, provider)}/${fileName}`, 'utf8'),
-  );
-}
+  function getDelay(chain) {
+    return data.CHAINS[chain].delay || DEFAULT_DELAY;
+  }
 
-function checkZeroBalance(balances: { [key: string]: string }) {
-  if (balances) {
-    Object.entries(balances).forEach(([key, value]) => {
-      if (Number(value) == 0) {
-        delete balances[key];
-      }
+  async function writeDataToFile(data, fileName, chain, provider) {
```

Comment:
As writing to file will be depreciated we could change the name of these functions to "writeDataToStorage" or similar